FG3 not downloading OSX natives

Appears to be a regression of the issue fixed by PR #424 and described here: https://github.com/MinecraftForge/MinecraftForge/issues/3964

Example error:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/grondag/MinecraftForge/build.gradle' line: 69

* What went wrong:
A problem occurred evaluating root project 'Forge'.
> A problem occurred configuring project ':clean'.
   > Could not resolve all files for configuration ':clean:downloadDeps0'.
      > Could not find java-objc-bridge-natives-osx.jar (ca.weblite:java-objc-bridge:1.0.0).
        Searched in the following locations:
            https://repo.maven.apache.org/maven2/ca/weblite/java-objc-bridge/1.0.0/java-objc-bridge-1.0.0-natives-osx.jar
```

--- a/src/common/java/net/minecraftforge/gradle/common/util/MinecraftRepo.java
+++ b/src/common/java/net/minecraftforge/gradle/common/util/MinecraftRepo.java
@@ -168,6 +168,7 @@ public class MinecraftRepo extends BaseRepo {
                 VersionJson meta = Utils.loadJson(json, VersionJson.class);
                 for (VersionJson.Library lib : meta.libraries) {
                     //TODO: Filter?
+                    builder.dependencies().add(lib.name, "compile");
                     if (lib.downloads.classifiers != null) {
                         if (lib.downloads.classifiers.containsKey("test")) {
                             builder.dependencies().add(lib.name, "test").withClassifier("test");
@@ -175,8 +176,6 @@ public class MinecraftRepo extends BaseRepo {
                         if (lib.natives != null && lib.natives.containsKey(CURRENT_OS) && !lib.getArtifact().getName().contains("java-objc-bridge")) {
                             builder.dependencies().add(lib.name, "runtime").withClassifier(lib.natives.get(CURRENT_OS));
                         }
-                    } else {
-                        builder.dependencies().add(lib.name, "compile");
                     }
                 }
             } else {

--- a/src/common/java/net/minecraftforge/gradle/common/util/MinecraftRepo.java
+++ b/src/common/java/net/minecraftforge/gradle/common/util/MinecraftRepo.java
@@ -168,14 +168,15 @@ public class MinecraftRepo extends BaseRepo {
                 VersionJson meta = Utils.loadJson(json, VersionJson.class);
                 for (VersionJson.Library lib : meta.libraries) {
                     //TODO: Filter?
-                    builder.dependencies().add(lib.name, "compile");
                     if (lib.downloads.classifiers != null) {
                         if (lib.downloads.classifiers.containsKey("test")) {
                             builder.dependencies().add(lib.name, "test").withClassifier("test");
                         }
-                        if (lib.natives != null && lib.natives.containsKey(CURRENT_OS)) {
+                        if (lib.natives != null && lib.natives.containsKey(CURRENT_OS) && !lib.getArtifact().getName().contains("java-objc-bridge")) {
                             builder.dependencies().add(lib.name, "runtime").withClassifier(lib.natives.get(CURRENT_OS));
                         }
+                    } else {
+                        builder.dependencies().add(lib.name, "compile");
                     }
                 }
             } else {

--- a/src/mcp/java/net/minecraftforge/gradle/mcp/MCPRepo.java
+++ b/src/mcp/java/net/minecraftforge/gradle/mcp/MCPRepo.java
@@ -179,6 +179,7 @@ public class MCPRepo extends BaseRepo {
                 VersionJson meta = Utils.loadJson(json, VersionJson.class);
                 for (VersionJson.Library lib : meta.libraries) {
                     //TODO: Filter?
+                    builder.dependencies().add(lib.name, "compile");
                     if (lib.downloads.classifiers != null) {
                         if (lib.downloads.classifiers.containsKey("test")) {
                             builder.dependencies().add(lib.name, "test").withClassifier("test");
@@ -186,8 +187,6 @@ public class MCPRepo extends BaseRepo {
                         if (lib.natives != null && lib.natives.containsKey(MinecraftRepo.CURRENT_OS) && !lib.getArtifact().getName().contains("java-objc-bridge")) {
                             builder.dependencies().add(lib.name, "runtime").withClassifier(lib.natives.get(MinecraftRepo.CURRENT_OS));
                         }
-                    } else {
-                        builder.dependencies().add(lib.name, "compile");
                     }
                 }
                 builder.dependencies().add("net.minecraft:client:" + getMCVersion(version), "compile").withClassifier("extra");

--- a/src/mcp/java/net/minecraftforge/gradle/mcp/MCPRepo.java
+++ b/src/mcp/java/net/minecraftforge/gradle/mcp/MCPRepo.java
@@ -179,14 +179,15 @@ public class MCPRepo extends BaseRepo {
                 VersionJson meta = Utils.loadJson(json, VersionJson.class);
                 for (VersionJson.Library lib : meta.libraries) {
                     //TODO: Filter?
-                    builder.dependencies().add(lib.name, "compile");
                     if (lib.downloads.classifiers != null) {
                         if (lib.downloads.classifiers.containsKey("test")) {
                             builder.dependencies().add(lib.name, "test").withClassifier("test");
                         }
-                        if (lib.natives != null && lib.natives.containsKey(MinecraftRepo.CURRENT_OS)) {
+                        if (lib.natives != null && lib.natives.containsKey(MinecraftRepo.CURRENT_OS) && !lib.getArtifact().getName().contains("java-objc-bridge")) {
                             builder.dependencies().add(lib.name, "runtime").withClassifier(lib.natives.get(MinecraftRepo.CURRENT_OS));
                         }
+                    } else {
+                        builder.dependencies().add(lib.name, "compile");
                     }
                 }
                 builder.dependencies().add("net.minecraft:client:" + getMCVersion(version), "compile").withClassifier("extra");

